beacon-chain/rpc/beaconv1/state.go

```diff
@@ -2,29 +2,166 @@ package beaconv1
 
 import (
 	"context"
-	"errors"
+	"strconv"
+	"time"
 
 	ptypes "github.com/gogo/protobuf/types"
+	"github.com/pkg/errors"
 	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/beacon-chain/state"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
 // GetGenesis retrieves details of the chain's genesis which can be used to identify chain.
 func (bs *Server) GetGenesis(ctx context.Context, _ *ptypes.Empty) (*ethpb.GenesisResponse, error) {
-	return nil, errors.New("unimplemented")
+	genesisTime := bs.GenesisTimeFetcher.GenesisTime()
+	var defaultGenesisTime time.Time
+	var gt *ptypes.Timestamp
+	var err error
+	if genesisTime == defaultGenesisTime {
```

Comment:
Why do you have to do this? ptypes.TimestampProto(0) returns an error?

Reply:
Context here: https://github.com/prysmaticlabs/prysm/pull/6031

Reply:
Please add comments. Otherwise it isn't clear to the reader why you are doing this.